sky/clouds/vast.py

```diff
@@ -0,0 +1,296 @@
+""" Vast Cloud. """
+
+import typing
+from typing import Dict, Iterator, List, Optional, Tuple, Union
+
+from sky import clouds
+from sky.clouds import service_catalog
+from sky.utils import resources_utils
+
+if typing.TYPE_CHECKING:
+    from sky import resources as resources_lib
+
+_CREDENTIAL_FILES = [
+    'config.toml',
+]
+
+
+@clouds.CLOUD_REGISTRY.register
+class Vast(clouds.Cloud):
+    """ Vast GPU Cloud
+
+    _REPR | The string representation for the Vast GPU cloud object.
+    """
+    _REPR = 'Vast'
+    _CLOUD_UNSUPPORTED_FEATURES = {
+        clouds.CloudImplementationFeatures.SPOT_INSTANCE:
+            ('Spot is not supported, as vast API does not implement spot.'),
+        clouds.CloudImplementationFeatures.MULTI_NODE:
+            ('Multi-node not supported yet, as the interconnection among nodes '
+             'are non-trivial on Vast.'),
+        ## TODO: These are different from the fluffy cloud
+        clouds.CloudImplementationFeatures.CUSTOM_DISK_TIER:
+            ('Customizing disk tier is not supported yet on Vast.'),
+        clouds.CloudImplementationFeatures.STORAGE_MOUNTING:
+            ('Mounting object stores is not supported on Vast. To read data '
+             'from object stores on Vast, use `mode: COPY` to copy the data '
+             'to local disk.'),
+    }
+    _MAX_CLUSTER_NAME_LEN_LIMIT = 120
```

Comment:
I tried to search this but does not seems yielding an answer for me. Is it possible to add some doc reference for this?

Reply:
right, that was just sloppiness on my part. I'll update it. Thanks.

Reply:
there was some additional cleanup required: https://github.com/kristopolous/skypilot/commit/3eb8823df98579ba8a9ef2bc1dc2983c8e3d3053

Reply:
bump for this - not seeing the reference now?

Reply:
I searched for it to.

```
aws.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 248
azure.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 42
cudo.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 60
fluidstack.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 57
gcp.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 35
ibm.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 47
kubernetes.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 42
lambda_cloud.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 57
oci.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 200
paperspace.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 120
runpod.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 120
scp.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 28
vast.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 120
vsphere.py:    _MAX_CLUSTER_NAME_LEN_LIMIT = 80  # The name can't exceeds 80 characters
```

zero other providers have a comment for it. That line is there to be consistent with the other 13 files. We don't actually have  a maximum length.

What do you want me to do?

Reply:
I mean this kind of comment that refers to the maximum length value:

https://github.com/skypilot-org/skypilot/blob/138679859b9844a8737f8dff1bf5a739e77e96c4/sky/clouds/aws.py#L110-L115

https://github.com/skypilot-org/skypilot/blob/138679859b9844a8737f8dff1bf5a739e77e96c4/sky/clouds/azure.py#L67-L72

https://github.com/skypilot-org/skypilot/blob/138679859b9844a8737f8dff1bf5a739e77e96c4/sky/clouds/gcp.py#L142-L150

If vast does not have a maximum length then it should be fine. Just want to figure out why the number is `120` instead of other numbers.